### PR TITLE
feat: Improved timeout err msg

### PR DESCRIPTION
**Pull request by sigmaaa**
## Description

Improved error message for server timeout error

Fixes # ([IEP-1063](https://jira.espressif.com:8443/browse/IEP-1063))

## Type of change

Please delete options that are not relevant.
- New feature (non-breaking change which adds functionality)

## How has this been tested?

- Test 1
set gdb server timeout to 1 sec -> run debug -> verify new error message dialog

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Debug

## Checklist
- [x] PR Self Reviewed
- [x] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

## Summary by CodeRabbit

- **New Features**
	- Enhanced user experience by introducing a dialog to display server timeout errors, providing guidance on adjusting timeout settings for improved reliability during OpenOCD server launches.

### CODE REVIEW

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/ui/messages.properties [hunk 1]
```diff
@@ -410,4 +410,9 @@ TabMain_Launch_Config=Launch Configuration:
...
 
 ## Console Messages ##
-OpenOCDConsole_ErrorGuideMessage=Please refer to the troubleshooting guide below to identify the problem.
\ No newline at end of file
+OpenOCDConsole_ErrorGuideMessage=Please refer to the troubleshooting guide below to identify the problem.
+
+## Timeout Exception Dialog ##
+ServerTimeoutErrorDialog.customAreaMessage=To increase timeout time visit <a>the Espressif Preference Page</a>.
+ServerTimeoutErrorDialog.message=Starting OpenOCD timed out. Try to increase the `GDB server launch timeout`
+ServerTimeoutErrorDialog.title=Problem Occured
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/ui/messages.properties]: fixed

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/LaunchConfigurationDelegate.java [hunk 2]
```diff
@@ -446,8 +448,22 @@ else if (sessionType == SessionType.CORE)
...
+				// Throwing exception with OK status to terminate launch sequence
+				throw new DebugException(new Status(IStatus.OK, GdbPlugin.PLUGIN_ID, DebugException.REQUEST_FAILED,
+						"Error in services launch sequence", e1.getCause())); //$NON-NLS-1$
+			}
+			else
+			{
+				throw new DebugException(new Status(IStatus.ERROR, GdbPlugin.PLUGIN_ID, DebugException.REQUEST_FAILED,
+						"Error in services launch sequence", e1.getCause())); //$NON-NLS-1$
+			}
+
```
[the remark this answers is not in the document]
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/LaunchConfigurationDelegate.java:466]: this message is not externalized in dependency, so it's ok to leave it like this for now

**Review by sigmaaa — COMMENTED** [review]
self reviewed

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/ui/ServerTimeoutErrorDialog.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			{
+				PreferencesUtil
+						.createPreferenceDialogOn(parent.getShell(), ESPRESSIF_PREFERENCES_MAINPAGE_ID, null, null)
+						.open();
+			}
+		});
+		;
+		return link;
+	}
+}
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/ui/ServerTimeoutErrorDialog.java:61]: It's better to have an internal class for this and override only createCustomArea() method.

```
MessageDialog dialog = new MessageDialog(Plugin.getActiveWorkbenchShell(), title, null /* default image */,
				message, MessageDialog.ERROR, new String[] { IDialogConstants.OK_LABEL }, 0) {
			@Override
			protected Control createCustomArea(Composite parent) {
				//implementation here
					}
				});
				
			}
		};
		dialog.open();
```

**Review by kolipakakondal — APPROVED** [review]
LGTM

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/LaunchConfigurationDelegate.java [hunk 4]
```diff
@@ -446,8 +448,22 @@ else if (sessionType == SessionType.CORE)
 		}
 		catch (ExecutionException e1)
 		{
-			throw new DebugException(new Status(IStatus.ERROR, GdbPlugin.PLUGIN_ID, DebugException.REQUEST_FAILED,
-					"Error in services launch sequence", e1.getCause())); //$NON-NLS-1$
+			if (e1.getMessage().contains("Starting OpenOCD timed out.")) //$NON-NLS-1$
+			{
+				Display.getDefault().asyncExec(() -> {
+					ServerTimeoutErrorDialog.openError(Display.getDefault().getActiveShell());
+
```
Comment (kolipakakondal) [inline bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/LaunchConfigurationDelegate.java:455]: See if you can propogate back this exception with appropriate message to the UI class and handle there

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/ui/ServerTimeoutErrorDialog.java (hunk 3 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/ui/ServerTimeoutErrorDialog.java]: done

File: bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/LaunchConfigurationDelegate.java (hunk 4 above)
Comment (sigmaaa) [reply bundles/com.espressif.idf.debug.gdbjtag.openocd/src/com/espressif/idf/debug/gdbjtag/openocd/dsf/LaunchConfigurationDelegate.java]: propagated via statusHandler in the latest commit

**Pull request merged**